server/node/src/main/java/org/kaaproject/kaa/server/admin/services/KaaAdminServiceImpl.java

```diff
@@ -560,32 +555,27 @@ public void deleteTenant(String userId) throws KaaAdminServiceException {
         }
     }
 
-    private void checkUserProfile(org.kaaproject.kaa.common.dto.admin.UserDto userDto) throws KaaAdminServiceException {
-        if (isEmpty(userDto.getUsername())) {
-            throw new IllegalArgumentException("Username is not valid.");
-        } else if (isEmpty(userDto.getFirstName())) {
+    private void checkUserProfile(UserProfileUpdateDto userProfileUpdateDto) throws KaaAdminServiceException {
+        if (isEmpty(userProfileUpdateDto.getFirstName())) {
             throw new IllegalArgumentException("First name is not valid.");
-        } else if (isEmpty(userDto.getLastName())) {
+        } else if (isEmpty(userProfileUpdateDto.getLastName())) {
             throw new IllegalArgumentException("Last name is not valid.");
-        } else if (isEmpty(userDto.getMail())) {
+        } else if (isEmpty(userProfileUpdateDto.getMail())) {
             throw new IllegalArgumentException("Mail is not valid.");
-        } else if (userDto.getAuthority() == null) {
-            throw new IllegalArgumentException("Authority is not valid.");
         }
     }
 
     @Override
-    public org.kaaproject.kaa.common.dto.admin.UserDto editUserProfile(org.kaaproject.kaa.common.dto.admin.UserDto userDto)
+    public void editUserProfile(UserProfileUpdateDto userProfileUpdateDto)
             throws KaaAdminServiceException {
+        checkAuthority(KaaAuthorityDto.TENANT_USER, KaaAuthorityDto.TENANT_DEVELOPER, KaaAuthorityDto.KAA_ADMIN, KaaAuthorityDto.TENANT_ADMIN);
```

Comment:
Are you sure that this is required? Since we already have following line in security configuration:
<security:intercept-url pattern="/kaaAdmin/rest/**" access="hasAnyAuthority('KAA_ADMIN','TENANT_ADMIN','TENANT_DEVELOPER','TENANT_USER')"/>
This step is not required. @ikulikov please provide your comments